ddl/constraint.go

```diff
@@ -86,15 +87,25 @@ func (w *worker) onAddCheckConstraint(d *ddlCtx, t *meta.Meta, job *model.Job) (
 	originalState := constraintInfoInMeta.State
 	switch constraintInfoInMeta.State {
 	case model.StateNone:
-		// none -> write only
 		job.SchemaState = model.StateWriteOnly
 		constraintInfoInMeta.State = model.StateWriteOnly
 		ver, err = updateVersionAndTableInfoWithCheck(d, t, job, tblInfo, originalState != constraintInfoInMeta.State)
 	case model.StateWriteOnly:
-		// write only -> public
-		err = w.verifyRemainRecordsForCheckConstraint(dbInfo, tblInfo, constraintInfoInMeta, job)
-		if err != nil {
-			return ver, errors.Trace(err)
+		job.SchemaState = model.StateWriteReorganization
+		constraintInfoInMeta.State = model.StateWriteReorganization
+		ver, err = updateVersionAndTableInfoWithCheck(d, t, job, tblInfo, originalState != constraintInfoInMeta.State)
+	case model.StateWriteReorganization:
```

Comment:
Why we add a new state for this operation？

Reply:
Because I want to make sure that the state of all tidb nodes is changed to `StateWriteOnly` before doing remaining data verification, so that no data is written that violates the check constraint.
 If doing stock data verification in `StateWriteOnly` state, the state of some nodes may be still in `StateNone `( am i right?), so even if the remaining data verification passes, there are still `StateNone` nodes that can write data that violates the check constraint. Hence I added a state to ensure that all nodes change at least to `StateWriteOnly` before verifying the data. Is this right or not, please give some advice, thks

Reply:
Ok, there was no PR description before, I don't know why it suddenly added a status, so ask.

Reply:
Seems here we missed delete-only state, that is before writeonly state.  write reorg means will reorg table data.